### ISSUE TITLE
Document PyUnicode_* API

### ISSUE THREAD

**Issue by avassalotti** [python/cpython#46236]
BPO | [1944](https://bugs.python.org/issue1944)
--- | :---
Nosy | @malemburg, @birkenfeld, @vstinner, @avassalotti, @berkerpeksag, @vadmium, @serhiy-storchaka, @shihai1991, @furkanonder
PRs | <li>python/cpython#20011</li>
Files | <li>[unicode.patch](https://bugs.python.org/file13717/unicode.patch "Uploaded as text/plain at 2009-04-18.15:18:44 by donlorenzo"): docs for PyUnicodes C-API functions: FromFormat, FromFormatV, FromString, FromStringAndSize, Partition, RPartition and RSplit</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2008-01-27.06:26:42.019>
labels = ['type-feature', 'docs']
title = 'Document PyUnicode_* API'
updated_at = <Date 2020-06-21.12:39:22.129>
user = 'https://github.com/avassalotti'
```

bugs.python.org fields:
```python
activity = <Date 2020-06-21.12:39:22.129>
actor = 'shihai1991'
assignee = 'docs@python'
closed = False
closed_date = None
closer = None
components = ['Documentation']
creation = <Date 2008-01-27.06:26:42.019>
creator = 'alexandre.vassalotti'
dependencies = []
files = ['13717']
hgrepos = []
issue_num = 1944
keywords = ['patch']
message_count = 9.0
messages = ['61734', '86116', '86121', '89100', '185552', '185725', '264571', '264575', '368582']
nosy_count = 11.0
nosy_names = ['lemburg', 'georg.brandl', 'vstinner', 'alexandre.vassalotti', 'donlorenzo', 'docs@python', 'berker.peksag', 'martin.panter', 'serhiy.storchaka', 'shihai1991', 'furkanonder']
pr_nums = ['20011']
priority = 'normal'
resolution = None
stage = 'patch review'
status = 'open'
superseder = None
type = 'enhancement'
url = 'https://bugs.python.org/issue1944'
versions = ['Python 3.5', 'Python 3.6']
```

</p></details>

[edit @encukou]: Converting the lists below to checkboxes:

- [x] PyUnicode_Resize
- [x] ~~PyUnicode_InternImmortal~~ (removed in 3.12)
- [x] PyUnicode_GetDefaultEncoding
- [x] ~~PyUnicode_SetDefaultEncoding~~ (removed in 3.2)
- [x] PyUnicode_BuildEncodingMap
- [x] PyUnicode_FromFormatV
- [x] PyUnicode_*UTF7*
- [x] PyUnicode_AsEncodedObject
- [x] PyUnicode_FromOrdinal
- [x] PyUnicode_DecodeFSDefault
- [x] PyUnicode_DecodeFSDefaultAndSize
- [x] PyUnicode_DecodeUTF7
- [x] PyUnicode_DecodeUTF7Stateful
- [x] PyUnicode_EncodeDecimal
- [x] PyUnicode_EncodeUTF7
- [x] PyUnicode_FromFormat
- [x] PyUnicode_FromString
- [x] PyUnicode_FromStringAndSize
- [x] PyUnicode_GetMax
- [x] PyUnicode_Partition
- [x] PyUnicode_RPartition
- [x] PyUnicode_RSplit
- [x] PyUnicode_IsIdentifier
- [x] PyUnicode_Append
- [x] PyUnicode_AppendAndDel
- [x] Py_UNICODE_REPLACEMENT_CHARACTER
- [x] PyUnicodeIter_Type
- [x] PyUnicode_AsDecodedObject
- [x] PyUnicode_AsDecodedUnicode
- [x] PyUnicode_AsEncodedUnicode
- [x] ~~_PyUnicode_ClearStaticStrings~~ (removed in 3.9)
- [x] ~~_PyUnicode_EQ~~ (removed in 3.14)
- [x] ~~_PyUnicode_FromId~~ (private)

### Linked PRs
* gh-127934
* gh-129961
* gh-129962
* gh-129966
* gh-130191
* gh-130335
* gh-130360
* gh-130361
* gh-130511
* gh-130512
* gh-130531
* gh-132040
* gh-132521
* gh-132628
* gh-132703
* gh-132706
* gh-132925
* gh-133149
* gh-133270
* gh-133769

**Comment by rruuaanng** [comment]
I have noticed that some tasks that have already been opened are not checked off. I believe we should select them.

**Comment by rruuaanng** [comment]
@encukou Let's remove the `PyUnicode_GetMax` entry, which was removed in `Misc/NEWS.d/3.10.0a1.rst`.

```
.. bpo: 41123
.. date: 2020-06-28-11-39-22
.. nonce: sjJWjQ
.. section: C API

Removed ``PyUnicode_GetMax()``.
```

**Comment by StanFromIreland** [comment]
`PyUnicode_EncodeDecimal` was removed in 3.11

**Comment by vstinner** [comment]
No, I  don't think that it's worth it to document them if they are going to be removed.

**Comment by StanFromIreland** [comment]
PyUnicode_FromFormatV is [documented](https://docs.python.org/3.13/c-api/unicode.html#c.PyUnicode_FromFormatV).

**Comment by StanFromIreland** [comment]
> No, I don't think that it's worth it to document them if they are going to be removed.

I agree, though they have been deprecated for many years and there is no planned removal date that I can find. Should we just remove them now?

**Comment by StanFromIreland** [comment]
The list needs to be updated, just `PyUnicode_BuildEncodingMap` is left.
